code/optimize/src/androidTest/java/com/adobe/marketing/mobile/optimize/OptimizeFunctionalTests.java

```diff
@@ -888,55 +1139,372 @@ public void call(Map<DecisionScope, Proposition> decisionScopePropositionMap) {
         Assert.assertFalse(propositionMap.containsKey(decisionScope2));
     }
 
-    //11
     @Test
-    public void testTrackPropositions_validPropositionInteractionsForDisplay() throws InterruptedException {
+    public void testGetPropositions_surfaceInCache() throws InterruptedException, IOException {
         //setup
         final Map<String, Object> configData = new HashMap<>();
         configData.put("edge.configId", "ffffffff-ffff-ffff-ffff-ffffffffffff");
         updateConfiguration(configData);
 
-        Offer offer = new Offer.Builder("xcore:personalized-offer:1111111111111111", OfferType.TEXT, "Text Offer!!").build();
-        Proposition proposition = new Proposition(
-                "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
-                Collections.singletonList(offer),
-                "eyJhY3Rpdml0eUlkIjoieGNvcmU6b2ZmZXItYWN0aXZpdHk6MTExMTExMTExMTExMTExMSIsInBsYWNlbWVudElkIjoieGNvcmU6b2ZmZXItcGxhY2VtZW50OjExMTExMTExMTExMTExMTEifQ==",
-                Collections.emptyMap()
-        );
-
-        //Action
+        //Call updatePropositions to update the personalizationRequestEventId
+        Optimize.updatePropositionsForSurfacePaths(Collections.singletonList("myView#htmlElement"), null, null);
```

Comment:
Can we not just provide a setter for the `requestEventId` and set it in the tests as opposed to updating each of the tests to invoke update propositions. We are doing the same in iOS functional tests.

Reply:
We would have to make `personalizationRequestEventId` in the `OptimizeExtension` class a static variable to expose it via a static setter to be used in the test. Not sure if we really want to do that, just to be able to test it

Reply:
Discussed offline and decided to keep it as is